src/providers/cryptoauthlib/key_operations.rs

```diff
@@ -84,4 +87,191 @@ impl Provider {
             }
         }
     }
+
+    pub(super) fn psa_import_key_internal(
+        &self,
+        app_name: ApplicationName,
+        op: psa_import_key::Operation,
+    ) -> Result<psa_import_key::Result> {
+        let key_name = op.key_name;
+        let key_triple = self.key_info_store.get_key_triple(app_name, key_name);
+        self.key_info_store.does_not_exist(&key_triple)?;
+
+        let key_attributes = op.attributes;
+        let key_type = match Provider::get_calib_key_type(&key_attributes) {
+            Ok(x) => x,
+            Err(error) => return Err(error),
+        };
+        let slot_id = match self.find_suitable_slot(&key_attributes) {
+            Ok(slot) => slot,
+            Err(error) => {
+                warn!("Failed to find suitable storage slot for key. {}", error);
+                return Err(error);
+            }
+        };
+        // There is no need check op.data.len() - ler the extraction functions do this.
+        let key_data = match extract_raw_key(key_attributes.key_type, &op.data) {
+            Ok(raw_key) => raw_key,
+            Err(error) => return Err(error),
+        };
+
+        let atca_error_status = self.device.import_key(key_type, &key_data, slot_id);
+
```

Comment:
I think here you could do a;
```rust
key_data.zeroize();
```
after the import to make sure the key data can not be read in memory after that.
Alternatively you can make the `extract_raw_key` function return a `Secret<Vec<u8>>` so that it's done automatically.

Reply:
I think changing the return value is the way to go.